docs/documentation/features.md

```diff
@@ -774,33 +774,60 @@ ConfigurationServiceProvider.overrideCurrent(overrider->overrider.withMetrics(me
 
 ### Micrometer implementation
 
-The micrometer implementation records a lot of metrics associated to each resource handled by the operator by default.
-In order to be efficient, the implementation removes meters associated with resources when they are deleted. Since it
-might be useful to keep these metrics around for a bit before they are deleted, it is possible to configure a delay
-before their removal. As this is done asynchronously, it is also possible to configure how many threads you want to
-devote to these operations. Both aspects are controlled by the `MicrometerMetrics` constructor so changing the defaults
-is a matter of instantiating `MicrometerMetrics` with the desired values and tell `ConfigurationServiceProvider` about
-it as shown above.
+The micrometer implementation is typically created using one of the provided factory methods which, depending on which
+is used, will return either a ready to use instance or a builder allowing users to customized how the implementation
```

Comment:
Having several entry points allows to provide often used configurations without users having to create them with the builder. It also allows to have stable API calls since these factory method implementations can be changed if we change the default behavior while the semantics of the method won't so I'd be in favor of keeping things like they are for now.